pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -57,11 +57,30 @@ var userRegistriesDir = filepath.FromSlash(".config/containers/registries.conf.d
 // The function errors if the newly created reference is not parsable.
 func (e *Endpoint) rewriteReference(ref reference.Named, prefix string) (reference.Named, error) {
 	refString := ref.String()
+	var newNamedRef string
 	if !refMatchesPrefix(refString, prefix) {
 		return nil, fmt.Errorf("invalid prefix '%v' for reference '%v'", prefix, refString)
 	}
-
-	newNamedRef := strings.Replace(refString, prefix, e.Location, 1)
+	if prefix[0:2] == "*." {
+		if strings.Contains(refString, "/") {
+			firstSlash := strings.Index(refString, "/")
```

Comment:
_Right now_, assuming `refMatchesPrefix` is true, wouldn’t it work to just split on first slash, and replace the domain with `e.Location` without any further processing?

---

Conceptually it would probably be cleanest for `refMatchesPrefix` to return the length of the matched prefix; then we wouldn’t even need to do a redundant comparison implied in `strings.Replace` and we could just splice strings — and more importantly it would naturally extend to a future version where `refMatchesPrefix` also returns the content matched by `*` so that it could be substituted.